analyses/cnv-comparison/util/cnv-comparison-functions.R

```diff
@@ -0,0 +1,171 @@
+# This script defines filtering and plotting functions to be sourced in 
+# `01-cnv-comparison-plotting.R`
+#
+# Chante Bethell for CCDL 2019
+# 
+# Usage:
+# This script is intended to be run via the command line from the top directory 
+# of the repository as follows:
+#
+# Rscript analyses/cnv-comparison/util/cnv-comparison-functions.R
+
+read_in_cnv <- function(file_path){
+  # Given the file path of the CNV data, read in the data.
+  #
+  # Arg:
+  #   file_path: file path of input data
+  #
+  # Return:
+  #   cnv_df: the data frame containing the input data
+
+  # Read in cnv data
+  cnv_df <-
+    read.table(gzfile(file_path), header = TRUE)
+
+  # Rearrange the columns
+  cnv_df <- cnv_df %>%
+    dplyr::select("chrom", "loc.start", "loc.end", "ID", "num.mark", "seg.mean")
+  
+  # Reorder chromosome levels 
+  levels(cnv_df$chrom) <- c(paste0("chr", c(1:22, "X", "Y")))
```

Comment:
The above doesn't actually convert to a factor, but makes it a kind of strange hybrid character with a levels attribute:
`dplyr::mutate(chrom = factor(chrom, levels = paste0("chr", c(1:22, "X", "Y"))`
should work

Reply:
If you look at the before and after plots, you can see that the order didn't actually change but the labels did. Which would be bad...

Reply:
Ahh, I thought it was strange when I initially tried to convert to a factor, what R claimed to already be a factor. Thanks, I'll give that a shot now.

Reply:
oooh... add stringsAsFactors=False to your read.table (or use readr...)

Reply:
I also made that realization a minute ago. That'll also be in this upcoming commit 👍